include/onnxruntime/core/framework/data_types.h

```diff
@@ -497,7 +550,81 @@ class SparseTensorType : public SparseTensorTypeBase {
  private:
   SparseTensorType() {
     using namespace data_types_internal;
-    TensorElementTypeSetter<elemT>::SetSparseTensorElementType(mutable_type_proto());
+    TensorElementTypeSetter<elemT>::SetSparseTensorElementType(this->mutable_type_proto());
+  }
```

Comment:
nit: why add the superfluous `this->`?

Reply:
Removed it. FWIW - TensorType has the superfluous call.